rust/Makefile.toml

```
@@ -0,0 +1,90 @@
+[tasks.format-rust]
+args = ["fmt"]
+command = "cargo"
+description = "Format rust sources files."
+install_crate = { rustup_component_name = "rustfmt" }
+
+[tasks.lint-rust-format]
+args = ["fmt", "--all", "--", "--check"]
+command = "cargo"
+description = "Check format of sources files."
+install_crate = { rustup_component_name = "rustfmt" }
+
+[tasks.lint-rust]
+args = [
+  "clippy",
+  "--workspace",
+  "--locked",
+  "--all-targets",
+  "--all-features",
+  "--",
+  "-D",
+  "clippy::all",
+  "-D",
+  "warnings",
+]
+command = "cargo"
+description = "Check lint of all sources files."
+install_crate = { rustup_component_name = "clippy" }
+
+[tasks.lint-toml]
+args = ["lint"]
+command = "taplo"
+description = "Check lint of all toml files."
+install_crate = { crate_name = "taplo-cli", binary = "taplo", test_arg = "--help" }
+
+[tasks.format-toml]
+args = ["fmt"]
+command = "taplo"
+description = "Format toml file"
+install_crate = { crate_name = "taplo-cli", binary = "taplo", test_arg = "--help" }
+
+[tasks.lint]
+dependencies = ["lint-rust-format", "lint-rust", "lint-toml"]
+
+[tasks.format]
+dependencies = ["format-rust", "format-toml"]
+
+[tasks.clean]
+args = ["clean"]
+command = "cargo"
+
+[tasks.build]
+args = ["build", "--release", "--locked"]
+command = "cargo"
+
+[tasks.build-coverage]
+args = ["build"]
+command = "cargo"
+description = "Compile the source code and create testable artifacts."
+env = { RUSTFLAGS = "-Cinstrument-coverage" }
+
+[tasks.test]
+args = ["test", "--tests", "--workspace", "--locked", "--all-features"]
+command = "cargo"
+dependencies = ["build-coverage"]
+description = "Run all unit tests."
+env = { LLVM_PROFILE_FILE = "default.profraw", RUST_BACKTRACE = 1 }
+install_crate = { rustup_component_name = "llvm-tools-preview" }
+
+[tasks.test-coverage]
+dependencies = ["test"]
+install_crate = { crate_name = "grcov" }
+script = '''
+grcov . --binary-path ./target/debug/ -s . -t lcov --branch --ignore-not-existing --ignore "*cargo*" -o coverage.lcov
+'''
+
+[tasks.publish-cosmos]
+args = ["publish", "--token", "${CARGO_API_TOKEN}", "--manifest-path", "cosmos_sdk_grpc_client/Cargo.toml"]
+command = "cargo"
+
+[tasks.publish-okp4]
+args = ["publish", "--token", "${CARGO_API_TOKEN}", "--manifest-path", "okp4_grpc_client/Cargo.toml"]
+command = "cargo"
+
+[tasks.publish]
+dependencies = ["publish-cosmos", "publish-okp4"]
```

You need also to add dependency on the generation before publish. (Sorry I forgot to add it in [my comment](https://github.com/okp4/okp4-cosmos-proto/pull/28#discussion_r931000598)) 
```suggestion
dependencies = ["generate-proto", "publish-cosmos", "publish-okp4"]
```

---

well isn't that handled in the build step ?

---

In the GitHub workflow ?

---

I do not think so, because `publish` is trigger on [tags push](https://github.com/okp4/okp4-cosmos-proto/pull/28/files#diff-551d1fcf87f78cc3bc18a7b332a4dc5d8773a512062df881c5aba28a6f5c48d7R117) and the `build` workflow [isn't triggered](https://github.com/okp4/okp4-cosmos-proto/pull/28/files#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721L3-L10) on tags push.

---

Oh right !
ok